data_diff/diff_tables.py

```diff
@@ -364,11 +364,23 @@ def _validate_and_adjust_columns(self, table1, table2):
                 lowest = min(col1, col2, key=attrgetter("precision"))
 
                 if col1.precision != col2.precision:
-                    logger.warn(f"Using reduced precision {lowest} for column '{c}'. Types={col1}, {col2}")
+                    logger.warning(f"Using reduced precision {lowest} for column '{c}'. Types={col1}, {col2}")
 
                 table1._schema[c] = col1.replace(precision=lowest.precision, rounds=lowest.rounds)
                 table2._schema[c] = col2.replace(precision=lowest.precision, rounds=lowest.rounds)
 
+            elif isinstance(col1, NumericType):
+                if not isinstance(col2, NumericType):
+                    raise TypeError(f"Incompatible types for column {c}:  {col1} <-> {col2}")
+
+                lowest = min(col1, col2, key=attrgetter("precision"))
+
+                if col1.precision != col2.precision:
+                    logger.warning(f"Using reduced precision {lowest} for column '{c}'. Types={col1}, {col2}")
+
+                table1._schema[c] = col1.replace(precision=lowest.precision)
+                table2._schema[c] = col2.replace(precision=lowest.precision)
```

Comment:
Can you explain why rounding isn't a problem for lower-precision floats and numerics, but is for DateTime? 👂🏻 

E.g. why do you have the problem where if you insert a datetime with `0.88888`, you run into a problem, but not with a `numeric(10, 1)`?

Reply:
It seems like all databases round their floats, instead of truncating them. Which makes sense as they all use the same FP algorithm.

But maybe we just didn't hit the right edge cases yet.